docs/powerpoint/insert-slides-into-presentation.md

```diff
@@ -0,0 +1,183 @@
+
+```javascript
+async function insertSlidesDestinationFormatting() {
+  await PowerPoint.run(async function(context) {
+    context.presentation
+    .insertSlidesFromBase64(chosenFileBase64,
+                            {
+                                formatting: "UseDestinationTheme",
+                                targetSlideId: "267#"
+                            }
+                          );
+    await context.sync();
+  });
+}
+```
+
+Of course, you typically won't know at coding time the ID or creation ID of the target slide. More commonly, an add-in will ask users to select the target slide. The following steps show how to get the ***nnn*#** ID of the currently selected slide and use it as the target slide.
+
+1. Create a function that gets the ID of the currently selected slide by using the [Office.context.document.getSelectedDataAsync](/javascript/api/office/office.document#getSelectedDataAsync_coercionType__callback_) method of the Common JavaScript APIs. The following is an example. Note that the call to `getSelectedDataAsync` is embedded in a Promise-returning function. It is a good practice to do this because the Promise-returning function is easy to call (and await) inside the `run()` method of an application-specific API. By awaiting the Promise-returning function, you are assured that the callback function of the Common API method has completed before the next line in the `run()` method executes.
+
+ 
+    ```javascript
+    function getSelectedSlideID() {
+      return new OfficeExtension.Promise<string>(function (resolve, reject) {
+        Office.context.document.getSelectedDataAsync(Office.CoercionType.SlideRange, function (asyncResult) {
+          try {
+            if (asyncResult.status === Office.AsyncResultStatus.Failed) {
+              reject(console.error(asyncResult.error.message));
+            } else {
+              resolve(asyncResult.value.slides[0].id);
+            }
+          }
+          catch (error) {
+            reject(console.log(error));
+          }
+        });
+      })
+    }
+    ```
+
+1. Call your new function inside the [PowerPoint.run()](/javascript/api/powerpoint#PowerPoint_run_batch_) of the main function and pass the ID that it returns (concatenated with the "#" symbol) as the value of the `targetSlideId` property of the `InsertSlideOptions` parameter. The following is an example.
+
+    ```javascript
+    async function insertAfterSelectedSlide() {
+        await PowerPoint.run(async function(context) {
+
+            const selectedSlideID = await getSelectedSlideID();
+
+            context.presentation.insertSlidesFromBase64(chosenFileBase64, {
+                formatting: "UseDestinationTheme",
+                targetSlideId: selectedSlideID + "#"
```

Comment:
This seems like a big problem with the APIs if users are expected to translate IDs to work with different APIs.